Allow /dev/mmcblk2boot0 usage in /etc/fw_env.config

Integrated since uboot-tools 2014.07.

Index: buildroot/package/uboot-tools/uboot-tools-2013.10-fw_env-switch-get-config-to-use-ms-in-sscanf.patch
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/uboot-tools/uboot-tools-2013.10-fw_env-switch-get-config-to-use-ms-in-sscanf.patch	2015-01-29 12:28:55.676698363 +0100
@@ -0,0 +1,76 @@
+From: Tom Rini <trini@ti.com>
+Date: Fri, 28 Mar 2014 16:03:33 +0000 (-0400)
+Subject: fw_env.c: Switch get_config to use '%ms' in sscanf
+X-Git-Tag: v2014.07-rc1~2^2~3^2~42
+X-Git-Url: http://git.denx.de/?p=u-boot.git;a=commitdiff_plain;h=229695fee97abc7f4adad92eb8e036792c144bc6
+
+fw_env.c: Switch get_config to use '%ms' in sscanf
+
+We currently limit ourself to 16 characters for the device name to read
+the environment from.  This is insufficient for /dev/mmcblk0boot1 to
+work for example.  Switch to '%ms' which gives us a dynamically
+allocated buffer instead.  We're short lived enough to not bother
+free()ing the buffer.
+
+Signed-off-by: Tom Rini <trini@ti.com>
+---
+
+diff --git a/tools/env/fw_env.c b/tools/env/fw_env.c
+index d228cc3..f5cd521 100644
+--- a/tools/env/fw_env.c
++++ b/tools/env/fw_env.c
+@@ -40,7 +40,7 @@
+ 	_min1 < _min2 ? _min1 : _min2; })
+ 
+ struct envdev_s {
+-	char devname[16];		/* Device name */
++	const char *devname;		/* Device name */
+ 	ulong devoff;			/* Device offset */
+ 	ulong env_size;			/* environment size */
+ 	ulong erase_size;		/* device erase size */
+@@ -1243,7 +1243,7 @@ static int parse_config ()
+ 		return -1;
+ 	}
+ #else
+-	strcpy (DEVNAME (0), DEVICE1_NAME);
++	DEVNAME (0) = DEVICE1_NAME;
+ 	DEVOFFSET (0) = DEVICE1_OFFSET;
+ 	ENVSIZE (0) = ENV1_SIZE;
+ 	/* Default values are: erase-size=env-size */
+@@ -1258,7 +1258,7 @@ static int parse_config ()
+ #endif
+ 
+ #ifdef HAVE_REDUND
+-	strcpy (DEVNAME (1), DEVICE2_NAME);
++	DEVNAME (1) = DEVICE2_NAME;
+ 	DEVOFFSET (1) = DEVICE2_OFFSET;
+ 	ENVSIZE (1) = ENV2_SIZE;
+ 	/* Default values are: erase-size=env-size */
+@@ -1297,6 +1297,7 @@ static int get_config (char *fname)
+ 	int i = 0;
+ 	int rc;
+ 	char dump[128];
++	char *devname;
+ 
+ 	fp = fopen (fname, "r");
+ 	if (fp == NULL)
+@@ -1307,8 +1308,8 @@ static int get_config (char *fname)
+ 		if (dump[0] == '#')
+ 			continue;
+ 
+-		rc = sscanf (dump, "%s %lx %lx %lx %lx",
+-			     DEVNAME (i),
++		rc = sscanf (dump, "%ms %lx %lx %lx %lx",
++			     &devname,
+ 			     &DEVOFFSET (i),
+ 			     &ENVSIZE (i),
+ 			     &DEVESIZE (i),
+@@ -1317,6 +1318,8 @@ static int get_config (char *fname)
+ 		if (rc < 3)
+ 			continue;
+ 
++		DEVNAME(i) = devname;
++
+ 		if (rc < 4)
+ 			/* Assume the erase size is the same as the env-size */
+ 			DEVESIZE(i) = ENVSIZE(i);
